tests/e2e-core-common/src/test/java/org/glassfish/jersey/tests/e2e/common/message/internal/HeaderUtilsTest.java

```diff
@@ -180,4 +184,97 @@ public void testAsHeaderString() throws Exception {
         final String result = HeaderUtils.asHeaderString(values, null);
         assertEquals("value,[null]," + uri.toASCIIString(), result);
     }
+
+    @Test
+    public void testgetPreferedNewCookie(){
+
+        Date earliestDate;
+        Date latestDate;
+
+        try {
+            earliestDate = HttpDateFormat.readDate("Fri, 1 Jan 2020 00:00:00 GMT");
```

Comment:
Please use Calendar or LocalDateTime to set the dates you want, to prevent parsing exception with foreign locales